header/serde.go

```diff
@@ -64,18 +64,30 @@ func UnmarshalExtendedHeader(data []byte) (*ExtendedHeader, error) {
 	return out, nil
 }
 
-// msgID computes an id for a pubsub message
+// unmarhsalCommit exists to assist the MsgID function in generating a unique
+// message ID without the additional allocations that the full
+// UnmarshalExtendedHeader would cause.
+func unmarshalCommit(data []byte) (*core.Commit, error) {
+	in := &header_pb.ExtendedHeader{}
+	err := in.Unmarshal(data)
+	if err != nil {
+		return nil, err
+	}
+
+	return core.CommitFromProto(in.Commit)
+}
+
+// MsgID computes an id for a pubsub message
 // TODO(@Wondertan): This cause additional allocations per each recvd message in the topic
```

Comment:
Already outdated? (assuming this and this changes https://github.com/celestiaorg/go-header/pull/251)

Reply:
There are two redundant allocs. The one in go-header fixes one and this is another one. The one here is not fixable entirely without large protocol breaking changes, so we only minimize its impact